Makefile

```diff
@@ -122,30 +128,39 @@ push-tagged-image: export TAG=snapshot-$(shell git branch --show-current | sed "
 push-tagged-image: push-image
 
 # Generate manifests e.g. CRD, RBAC etc.
-manifests: manifests-k8s manifests-ocp
+manifests: prepare-manifests-directory manifests-k8s manifests-ocp strip-helm-labels
 
-manifests-k8s: manifests-crd manifests-k8s-csidriver
-	cp "$(KUBERNETES_CRD_AND_OTHERS_YAML)" "$(KUBERNETES_OLM_YAML)"
-	cat "$(KUBERNETES_CRD_AND_OTHERS_YAML)" "$(KUBERNETES_CSIDRIVER_YAML)" > "$(KUBERNETES_ALL_YAML)"
+strip-helm-labels:
+	./hack/build/strip-helm-labels.sh \
+		$(KUBERNETES_OLM_YAML) \
+		$(KUBERNETES_ALL_YAML) \
+		$(OPENSHIFT_OLM_YAML) \
+		$(OPENSHIFT_ALL_YAML)
 
-manifests-crd: generate-crd controller-gen kustomize
-	# Create directories for manifests if they do not exist
-	mkdir -p config/deploy/kubernetes
+prepare-manifests-directory:
+	find $(MANIFESTS_DIR) -type f -not -name 'kustomization.yaml' -delete
 
-	# Generate kubernetes.yaml
+manifests-k8s: manifests-k8s-crd manifests-k8s-csidriver
+	cp "$(KUBERNETES_CORE_YAML)" "$(KUBERNETES_OLM_YAML)"
+	cat "$(KUBERNETES_CORE_YAML)" "$(KUBERNETES_CSIDRIVER_YAML)" > "$(KUBERNETES_ALL_YAML)"
+
+manifests-k8s-crd: prepare-crd-for-helm kustomize
```

Comment:
there shouldn't be a separate crd step for k8s and ocp. Since this seems to create more than just the crd, maybe the step is just named incorrectly